core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -109,48 +126,65 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
           else reply ! Committed(offsets.asScala.toMap)
         }
       })
-      //right now we can not store commits in consumer - https://issues.apache.org/jira/browse/KAFKA-3412
-      poll()
+
     case Subscribe(topics, listener) =>
       consumer.subscribe(topics.toList.asJava, new WrappedAutoPausedListener(consumer, listener))
+
     case SubscribePattern(pattern, listener) =>
       consumer.subscribe(Pattern.compile(pattern), new WrappedAutoPausedListener(consumer, listener))
+
     case Poll(target) =>
-      if (target == this) {
-        poll()
-        currentPollTask = schedulePollTask()
-      }
+      if (target == this) poll()
       else {
         // Message was enqueued before a restart - can be ignored
         log.debug("Ignoring Poll message with stale target ref")
       }
+      schedulePollTask()
 
-    case req: RequestMessages =>
+    case sub: SubscriptionMessage =>
       context.watch(sender())
-      checkOverlappingRequests("RequestMessages", sender(), req.topics)
-      requests = requests.updated(sender(), req)
-      poll()
```

Comment:
Wouldn't a simple and efficient solution be to not perform the `poll` immediately and instead send an `ExtraPoll` message to self that will do the `poll`. In that way additional `RequestMessages` that are pending in the mailbox will be received first and the `poll` will process all of them. A short delay could also be used to collect more `RequestMessages` before the first `poll`.

It could even be adaptive so that it still performs it immediately if there is only one known subscriber.